src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -186,27 +190,56 @@ protected virtual RecordValue Find(RecordValue baseRecord)
             return null;
         }
 
-        protected static bool Matches(RecordValue currentRecord, RecordValue baseRecord)
+        protected static bool Matches(RecordValue currentRecord, RecordValue baseRecord, bool exactly = false)
         {
-            foreach (var field in baseRecord.Fields)
+            var ret = true;
+
+            var currentRecordEnumerator = currentRecord.Fields.GetEnumerator();
+            var baseRecordEnumerator = baseRecord.Fields.GetEnumerator();
+
+            while (baseRecordEnumerator.MoveNext())
```

Comment:
> while (baseRecordEnumerator.MoveNext()) [](http://example.com/codeflow?start=11&length=40)

can we use foreach keyword  instead?

Reply:
This was implement to work with the EXACTLY argument. If EXACTLY arg is true, both currentRecordEnumerator and baseRecordEnumerator should have no elements left to compare. IMO is an easy way to track any left element.